tests/MessageTest.php

```diff
@@ -48,11 +48,12 @@ public function testEncodingQuotedPrintable()
         $raw = "Subject: ESPAÑA\r\n"
             . "Date: =?ISO-8859-2?Q?Fri,_13_Jun_2014_17:18:44_+020?= =?ISO-8859-2?Q?0_(St=F8edn=ED_Evropa_(letn=ED_=E8as))?=\r\n"
             . "Content-Type: multipart/alternative; boundary=\"$boundary\"\r\n"
+            . "\r\n"
             . "--$boundary\r\n"
             . "Content-Transfer-Encoding: quoted-printable\r\n"
             . "Content-Type: text/html; charset=\"windows-1252\"\r\n"
             . "\r\n"
-            . "<html><body>Espa=F1a</body></html>\r\n\r\n"
+            . "<html><body>Espa=C3=B1a</body></html>\r\n\r\n"
```

Comment:
Why did you change this? The original text was received in a bug report.

Reply:
This is the change I was least comfortable with.

Even though `F1` is the right UTF-8 code for n with tilde, I was unable to reproduce in a real email the character written like this. Instead every mail agent uses =C3=B1.

I think we need to refactor test assets as a whole:

1. Write emails within PHP in pure UTF-8
1. Use an external tool/library/ext to convert them into different charsets
1. Send them to Dovecot (already done)
1. Test this library reads the result in UTF-8 (already done)

String concatenation seems too week to handle the encoding topic properly (in the tests).

Reply:
Fair enough. We can use ddeboer/transcoder (or some other lib) for step 2.

Reply:
Reminder: https://github.com/ddeboer/imap/issues/176

It have to said that https://github.com/ddeboer/imap/commit/21f3817c674062d275537a7a818ba26e4e9962e9 behavied completely differently from now; something somewhere got broken and testing against Gmail didn't help.